app/controllers/adoption_application_controller.rb

```diff
@@ -0,0 +1,47 @@
+class AdoptionApplicationController < ApplicationController
+
+   def show
+      @adoption_app = AdoptionApplication.find(params[:id])
+      
+      @pets = Pet.search(params[:search])
+
+      if params[:add_pet].present?
```

Comment:
This show action is doing **a lot** of different things, and you could break it up into a couple different actions instead in order to better comply with the Single Responsibility Principle and REST. 

When a user clicks on "Add Pet", the operation that will lead to is the creation of a joins record in your adoption_application_pets table. Since this is a `creation` operation, it's best for this to go to a `create` action. But, you're not creating new application, so it shouldn't go to a `create` action in this controller. Instead, you can create the joins controller I mentioned above. You could have an `AdoptionApplicationPetsController` with a `create` action, which can handle the creation of this joins record, and then redirect back to this show page